ospray_modules/opendeck/ispc/cylindricStereoTracked/CylindricStereoTrackedCamera.ispc

```diff
@@ -37,28 +43,42 @@ void CylindricStereoTrackedCamera_initRay(uniform Camera* uniform _self,
 
     float eyeDelta = 0.0f;
     float alpha = 0.0f;
-    if (self->side_by_side)
+    if (self->bufferId == leftWall)
     {
-        if (screen.x < 0.5f)
-        {
-            eyeDelta = -self->interpupillaryDistance * 0.5f;
-            alpha = 2.0f * screen.x * pi;
-        }
-        else
-        {
-            eyeDelta = self->interpupillaryDistance * 0.5f;
-            alpha = 2.0f * (screen.x - 0.5f) * pi;
-        }
+        eyeDelta = -self->interpupillaryDistance * 0.5f;
+        alpha = screen.x * pi;
     }
-    else
+    else if (self->bufferId == rightWall)
     {
+        eyeDelta = self->interpupillaryDistance * 0.5f;
         alpha = screen.x * pi;
     }
+    else if (self->bufferId == leftFloor)
+    {
+        eyeDelta = -self->interpupillaryDistance * 0.5f;
+    }
+    else if (self->bufferId == rightFloor)
+    {
+        eyeDelta = self->interpupillaryDistance * 0.5f;
+    }
 
     vec3f pixelPos;
-    pixelPos.x = OPENDECK_RADIUS * -cosf(alpha);
-    pixelPos.y = OPENDECK_HEIGHT * screen.y;
-    pixelPos.z = OPENDECK_RADIUS * -sinf(alpha);
+    if (self->bufferId == leftWall || self->bufferId == rightWall)
+    {
+        pixelPos.x = OPENDECK_RADIUS * -cosf(alpha);
+        // pixelPos.y = OPENDECK_METALSTRIPE_HEIGHT + OPENDECK_HEIGHT *
+        // screen.y;
+        pixelPos.y = OPENDECK_METALSTRIPE_HEIGHT + OPENDECK_HEIGHT -
+                     OPENDECK_HEIGHT * screen.y;
+        pixelPos.z = OPENDECK_RADIUS * -sinf(alpha);
+    }
+    else if (self->bufferId == leftFloor || self->bufferId == rightFloor)
+    {
+        pixelPos.x = 2.0f * OPENDECK_RADIUS * (screen.x - 0.5f);
+        pixelPos.y = 0.0f;
+        pixelPos.z = -OPENDECK_RADIUS + OPENDECK_RADIUS * screen.y;
+        // pixelPos.z = -OPENDECK_RADIUS * screen.y;
```

Comment:
that's the top-left vs bottom-left thing?

Reply:
yes